### PR TITLE
Using pluggy to allow plug-able can interfaces.

### PR DESCRIPTION
**Pull request by pierreluctg**
Allows user of python-can to create their own plug-able interface without the need to modify the python-can source.

- [x] Add documentation

### CODE REVIEW

File: can/plugin.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+from pluggy import PluginManager
+
+from can.interfaces import hookspecs
```
Comment (hardbyte) [inline can/plugin.py]: There doesn't appear to be a hookspecs?

File: can/util.py [hunk 2]
```diff
@@ -4,8 +4,11 @@
 """
 from __future__ import absolute_import
 
+from pluggy import PluginManager
+
 import can
-from can.interfaces import VALID_INTERFACES
+from can.interfaces import VALID_INTERFACES, hookspecs
```
Comment (hardbyte) [inline can/util.py]: Missing import (`hookspecs`) breaks the build - https://travis-ci.org/hardbyte/python-can/jobs/328610852

**Comment by pierreluctg** [comment]
Hi @hardbyte, I have simplified the plugin interface and add documentation.

**Comment by pierreluctg** [comment]
Hi @hardbyte, is there anything else you would like to see changed in this PR?

Thank you

**Pull request merged**